### PR TITLE
No Database selection project generation issue fix

### LINKED ISSUE #15507: Gateway does not compile when no database is selected
##### **Overview of the issue**

I am trying to create a API gateway application based off Spring Cloud API gateway. However, upon running the command "mvnw", the build succeeded with the following compilation error.

**Error thrown**
```
2021-06-29 17:40:42.157  WARN 22184 --- [  restartedMain] onfigReactiveWebServerApplicationContext : Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'gatewaySwaggerResourcesProvider' defined in file [D:\projects\gatewayapp\target\classes\com\gwa\config\apidocs\GatewaySwaggerResourcesProvider.class]: Unsatisfied dependency expressed through constructor parameter 1; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'inMemorySwaggerResourcesProvider' defined in URL [jar:file:/C:/Users/onezerofive/.m2/repository/io/springfox/springfox-swagger-common/3.0.0/springfox-swagger-common-3.0.0.jar!/springfox/documentation/swagger/web/InMemorySwaggerResourcesProvider.class]: Bean instantiation via constructor failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [springfox.documentation.swagger.web.InMemorySwaggerResourcesProvider]: Constructor threw exception; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'openAPISpringfoxApiDocket' defined in class path resource [tech/jhipster/config/apidoc/SpringfoxAutoConfiguration.class]: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [springfox.documentation.spring.web.plugins.Docket]: Factory method 'openAPISpringfoxApiDocket' threw exception; nested exception is java.lang.NoClassDefFoundError: org/springframework/data/domain/Pageable
2021-06-29 17:40:42.223 ERROR 22184 --- [  restartedMain] o.s.boot.SpringApplication               : Application run failed

org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'gatewaySwaggerResourcesProvider' defined in file [D:\projects\gatewayapp\target\classes\com\gwa\config\apidocs\GatewaySwaggerResourcesProvider.class]: Unsatisfied dependency expressed through constructor parameter 1; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'inMemorySwaggerResourcesProvider' defined in URL [jar:file:/C:/Users/onezerofive/.m2/repository/io/springfox/springfox-swagger-common/3.0.0/springfox-swagger-common-3.0.0.jar!/springfox/documentation/swagger/web/InMemorySwaggerResourcesProvider.class]: Bean instantiation via constructor failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [springfox.documentation.swagger.web.InMemorySwaggerResourcesProvider]: Constructor threw exception; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'openAPISpringfoxApiDocket' defined in class path resource [tech/jhipster/config/apidoc/SpringfoxAutoConfiguration.class]: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [springfox.documentation.spring.web.plugins.Docket]: Factory method 'openAPISpringfoxApiDocket' threw exception; nested exception is java.lang.NoClassDefFoundError: org/springframework/data/domain/Pageable
        at org.springframework.beans.factory.support.ConstructorResolver.createArgumentArray(ConstructorResolver.java:800)
        at org.springframework.beans.factory.support.ConstructorResolver.autowireConstructor(ConstructorResolver.java:229)
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.autowireConstructor(AbstractAutowireCapableBeanFactory.java:1354)
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1204)
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:564)
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:524)
        at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:335)
        at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234)
        at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:333)
        at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:208)
        at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:944)
        at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:918)
        at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:583)
        at org.springframework.boot.web.reactive.context.ReactiveWebServerApplicationContext.refresh(ReactiveWebServerApplicationContext.java:63)
        at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:771)
        at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:763)
        at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:438)
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:339)
        at com.GatewayappApp.main(GatewayappApp.java:68)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at org.springframework.boot.devtools.restart.RestartLauncher.run(RestartLauncher.java:49)
Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'inMemorySwaggerResourcesProvider' defined in URL [jar:file:/C:/Users/onezerofive/.m2/repository/io/springfox/springfox-swagger-common/3.0.0/springfox-swagger-common-3.0.0.jar!/springfox/documentation/swagger/web/InMemorySwaggerResourcesProvider.class]: Bean instantiation via constructor failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [springfox.documentation.swagger.web.InMemorySwaggerResourcesProvider]: Constructor threw exception; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'openAPISpringfoxApiDocket' defined in class path resource [tech/jhipster/config/apidoc/SpringfoxAutoConfiguration.class]: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [springfox.documentation.spring.web.plugins.Docket]: Factory method 'openAPISpringfoxApiDocket' threw exception; nested exception is java.lang.NoClassDefFoundError: org/springframework/data/domain/Pageable
        at org.springframework.beans.factory.support.ConstructorResolver.instantiate(ConstructorResolver.java:315)
        at org.springframework.beans.factory.support.ConstructorResolver.autowireConstructor(ConstructorResolver.java:296)
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.autowireConstructor(AbstractAutowireCapableBeanFactory.java:1354)
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1204)
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:564)
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:524)
        at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:335)
        at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234)
        at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:333)
        at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:208)
        at org.springframework.beans.factory.config.DependencyDescriptor.resolveCandidate(DependencyDescriptor.java:276)
        at org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:1380)
        at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveDependency(DefaultListableBeanFactory.java:1300)
        at org.springframework.beans.factory.support.ConstructorResolver.resolveAutowiredArgument(ConstructorResolver.java:887)
        at org.springframework.beans.factory.support.ConstructorResolver.createArgumentArray(ConstructorResolver.java:791)
        ... 23 common frames omitted
Caused by: org.springframework.beans.BeanInstantiationException: Failed to instantiate [springfox.documentation.swagger.web.InMemorySwaggerResourcesProvider]: Constructor threw exception; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'openAPISpringfoxApiDocket' defined in class path resource [tech/jhipster/config/apidoc/SpringfoxAutoConfiguration.class]: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [springfox.documentation.spring.web.plugins.Docket]: Factory method 'openAPISpringfoxApiDocket' threw exception; nested exception is java.lang.NoClassDefFoundError: org/springframework/data/domain/Pageable
        at org.springframework.beans.BeanUtils.instantiateClass(BeanUtils.java:225)
        at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:117)
        at org.springframework.beans.factory.support.ConstructorResolver.instantiate(ConstructorResolver.java:311)
        ... 37 common frames omitted
Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'openAPISpringfoxApiDocket' defined in class path resource [tech/jhipster/config/apidoc/SpringfoxAutoConfiguration.class]: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [springfox.documentation.spring.web.plugins.Docket]: Factory method 'openAPISpringfoxApiDocket' threw exception; nested exception is java.lang.NoClassDefFoundError: org/springframework/data/domain/Pageable
        at org.springframework.beans.factory.support.ConstructorResolver.instantiate(ConstructorResolver.java:658)
        at org.springframework.beans.factory.support.ConstructorResolver.instantiateUsingFactoryMethod(ConstructorResolver.java:638)
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.instantiateUsingFactoryMethod(AbstractAutowireCapableBeanFactory.java:1334)
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1177)
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:564)
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:524)
        at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:335)
        at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234)
        at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:333)
        at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:208)
        at org.springframework.context.support.AbstractApplicationContext.getBean(AbstractApplicationContext.java:1154)
        at org.springframework.plugin.core.support.AbstractTypeAwareSupport$BeansOfTypeTargetSource.lambda$getBeansOfTypeExcept$1(AbstractTypeAwareSupport.java:224)
        at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
        at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:177)
        at java.base/java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:948)
        at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
        at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
        at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)
        at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
        at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578)
        at org.springframework.plugin.core.support.AbstractTypeAwareSupport$BeansOfTypeTargetSource.getBeansOfTypeExcept(AbstractTypeAwareSupport.java:225)
        at org.springframework.plugin.core.support.AbstractTypeAwareSupport$BeansOfTypeTargetSource.getTarget(AbstractTypeAwareSupport.java:204)
        at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:195)
        at com.sun.proxy.$Proxy137.stream(Unknown Source)
        at org.springframework.plugin.core.PluginRegistrySupport.initialize(PluginRegistrySupport.java:78)
        at org.springframework.plugin.core.OrderAwarePluginRegistry.initialize(OrderAwarePluginRegistry.java:216)
        at org.springframework.plugin.core.PluginRegistrySupport.getPlugins(PluginRegistrySupport.java:60)
        at org.springframework.plugin.core.SimplePluginRegistry.getPlugins(SimplePluginRegistry.java:100)
        at springfox.documentation.spring.web.plugins.DocumentationPluginsManager.documentationPlugins(DocumentationPluginsManager.java:105)
        at springfox.documentation.swagger.web.InMemorySwaggerResourcesProvider.<init>(InMemorySwaggerResourcesProvider.java:62)
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
        at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
        at org.springframework.beans.BeanUtils.instantiateClass(BeanUtils.java:212)
        ... 39 common frames omitted
Caused by: org.springframework.beans.BeanInstantiationException: Failed to instantiate [springfox.documentation.spring.web.plugins.Docket]: Factory method 'openAPISpringfoxApiDocket' threw exception; nested exception is java.lang.NoClassDefFoundError: org/springframework/data/domain/Pageable
        at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:185)
        at org.springframework.beans.factory.support.ConstructorResolver.instantiate(ConstructorResolver.java:653)
        ... 73 common frames omitted
Caused by: java.lang.NoClassDefFoundError: org/springframework/data/domain/Pageable
        at tech.jhipster.config.apidoc.customizer.JHipsterSpringfoxCustomizer.customize(JHipsterSpringfoxCustomizer.java:94)
        at tech.jhipster.config.apidoc.SpringfoxAutoConfiguration.lambda$openAPISpringfoxApiDocket$0(SpringfoxAutoConfiguration.java:113)
        at java.base/java.util.ArrayList.forEach(ArrayList.java:1540)
        at tech.jhipster.config.apidoc.SpringfoxAutoConfiguration.openAPISpringfoxApiDocket(SpringfoxAutoConfiguration.java:113)
        at tech.jhipster.config.apidoc.SpringfoxAutoConfiguration$$EnhancerBySpringCGLIB$$5a9fc64b.CGLIB$openAPISpringfoxApiDocket$3(<generated>)
        at tech.jhipster.config.apidoc.SpringfoxAutoConfiguration$$EnhancerBySpringCGLIB$$5a9fc64b$$FastClassBySpringCGLIB$$92e637e7.invoke(<generated>)
        at org.springframework.cglib.proxy.MethodProxy.invokeSuper(MethodProxy.java:244)
        at org.springframework.context.annotation.ConfigurationClassEnhancer$BeanMethodInterceptor.intercept(ConfigurationClassEnhancer.java:331)
        at tech.jhipster.config.apidoc.SpringfoxAutoConfiguration$$EnhancerBySpringCGLIB$$5a9fc64b.openAPISpringfoxApiDocket(<generated>)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:154)
        ... 74 common frames omitted
Caused by: java.lang.ClassNotFoundException: org.springframework.data.domain.Pageable
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:583)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
        ... 88 common frames omitted
```

##### **Motivation for or Use Case**

The error states that it cannot find the CLASS file for "org.springframework.data.domain.Pageable". Which is indeed true, I couldn't find any JAR or CLASS following that pathname.

However, I checked it up online, it seems that this particular CLASS file is part of a JAR under "org.springframework.data.spring-data-commons". As a matter of fact, I have that JAR and within it contains the problematic CLASS file.

##### **Reproduce the error**

I followed [this](https://www.youtube.com/watch?v=oNagVFsfBqY) tutorial on setting up a Spring Cloud API gateway.

Files edited/created
1. Commented away these related lines in _WebConfigurer.java_ in ... src/main/java/com/gwa/config
... ReactivePageableHandlerMethodArgumentResolver
... ReactiveSortHandlerMethodArgumentResolver

*prior to step 1
This compilation issue came about 
`src/main/java/com/mycompany/myapp/config/WebConfigurer.java:[13,36] package org.springframework.data.web does not exist`

I looked into the WebConfigurer.java file, there were some comment pointing to a function mentioning "TODO: remove when this is supported in spring-boot"... thus, I commented away and the build progressed till the current juncture of this post

2. Created a _main.java_ file in ... src/main/java/com/gwa/web/rest
3. Edited the _application-dev.yml_ and _application-prd.yml_ in ... src/main/resources/config

##### **Related issues**

NA

##### **Suggest a Fix**

As highlighted by the "caused by". I believe the solution, is to modify the file that dictates the errored path and re-point to the correct dependency.

##### **JHipster Version(s)**

Currently using 7.1 but this issue occurred in 7.0.1 too.

##### **JHipster configuration**

**jhipster info**
INFO! Using JHipster version installed locally in current project's node_modules
Welcome to the JHipster Information Sub-Generator

##### **JHipster Version(s)**

```
gatewayapp@0.0.1-SNAPSHOT D:\projects\gatewayapp
`-- (empty)

```

##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**

<details>
<summary>.yo-rc.json file</summary>
<pre>
{
  "generator-jhipster": {
    "blueprints": [],
    "otherModules": [],
    "applicationType": "gateway",
    "baseName": "gatewayapp",
    "jhipsterVersion": "7.1.0",
    "skipClient": false,
    "skipServer": false,
    "skipUserManagement": true,
    "skipCheckLengthOfIdentifier": false,
    "skipFakeData": false,
    "jhiPrefix": "jhi",
    "entitySuffix": "",
    "dtoSuffix": "DTO",
    "testFrameworks": [],
    "pages": [],
    "creationTimestamp": 1623998011866,
    "serviceDiscoveryType": "eureka",
    "reactive": true,
    "authenticationType": "jwt",
    "packageName": "com.gwa",
    "serverPort": "8088",
    "cacheProvider": "no",
    "enableHibernateCache": false,
    "databaseType": "no",
    "buildTool": "maven",
    "serverSideOptions": [],
    "websocket": false,
    "searchEngine": false,
    "messageBroker": false,
    "enableSwaggerCodegen": false,
    "clientFramework": "angularX",
    "withAdminUi": true,
    "clientTheme": "none",
    "enableTranslation": true,
    "nativeLanguage": "en",
    "packageFolder": "com/gwa",
    "jwtSecretKey": "YourJWTSecretKeyWasReplacedByThisMeaninglessTextByTheJHipsterInfoCommandForObviousSecurityReasons",
    "devDatabaseType": "no",
    "prodDatabaseType": "no",
    "clientPackageManager": "npm",
    "clientThemeVariant": "",
    "languages": ["en"],
    "devServerPort": 4200
  }
}

</pre>
</details>

##### **JDL for the Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<details>
<summary>JDL entity definitions</summary>

<pre>

</pre>
</details>

##### **Environment and Tools**

openjdk version "11.0.3" 2019-04-16
OpenJDK Runtime Environment AdoptOpenJDK (build 11.0.3+7)
OpenJDK 64-Bit Server VM AdoptOpenJDK (build 11.0.3+7, mixed mode)

node: v14.17.0

npm: 6.14.13

No change to package.json was detected. No package manager install will be executed.
Congratulations, JHipster execution is complete!
Sponsored with ❤️  by @oktadev.

##### **Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

NA

##### **Browsers and Operating System**

Browser: Chrome
Operating System: Windows 10

- [x] Checking this box is mandatory (this is just to show you read everything)

### PR DESCRIPTION
**Pull request by yuvarajsanjeevi**
Fix #15507

---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [ ] The JDL part is updated if necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (bellow reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

**Comment by mshima** [comment]
This PR looks good, but this will not close the issue because it doesn't fixes the problems outlined by @mraible in the daily builds.
I've triggered a manual build with this branch at https://github.com/hipster-labs/jhipster-daily-builds/actions/runs/1141429995.

And since we are going modular, we should add a no-database test to main generator.

**Comment by mshima** [comment]
@yuvarajsanjeevi do you want to keep working on the PR or should I change the description and merge?

**Comment by yuvarajsanjeevi** [comment]
@mshima 

I could see the parameters in the generated daily build

JHI_LIB_REPO=https://github.com/jhipster/jhipster-bom.git
JHI_LIB_BRANCH=auto
JHI_LIB_HOME=/home/runner/work/jhipster-daily-builds/jhipster-daily-builds/jhipster-bom
JHI_GEN_REPO=https://github.com/yuvarajsanjeevi/generator-jhipster.git
**JHI_GEN_BRANCH=main**
JHI_FOLDER_APP=/home/runner/work/jhipster-daily-builds/jhipster-daily-builds/app
JHI_JDL_APP=
JHI_APP=ngx-nodatabase-jwt


But the fix is in the branch fix#15507. Kindly run the daily build on this branch please

**Comment by mshima** [comment]
You are right https://github.com/hipster-labs/jhipster-daily-builds/actions/runs/1142420695.
Can you add gateway tests to daily builds to cover the original issue?

**Comment by yuvarajsanjeevi** [comment]
Do you mean adding new samples for gateway with no db in https://github.com/hipster-labs/jhipster-daily-builds and defining workflow in the no-database.yaml. Please confirm

**Comment by mshima** [comment]
> Do you mean adding new samples for gateway with no db in https://github.com/hipster-labs/jhipster-daily-builds and defining workflow in the no-database.yaml

Yes please.

**Comment by yuvarajsanjeevi** [comment]
Raised PR https://github.com/hipster-labs/jhipster-daily-builds/pull/124

**Comment by mshima** [comment]
https://github.com/hipster-labs/jhipster-daily-builds/actions/runs/1152758979

File: generators/server/templates/build.gradle.ejs [hunk 1]
```diff
@@ -542,6 +542,9 @@ dependencies {
     liquibaseRuntime "org.liquibase.ext:liquibase-mssql:<%= LIQUIBASE_VERSION %>"
     <%_ } _%>
     implementation "org.mapstruct:mapstruct:${mapstructVersion}"
+    <%_ if (reactive) { _%>
+    implementation "org.springframework.data:spring-data-commons"
+    <%_ } _%>
```
Comment (mshima) [inline generators/server/templates/build.gradle.ejs:547]: IMO spring-data should not be required without a database.
Where it's used?

Comment (yuvarajsanjeevi) [reply generators/server/templates/build.gradle.ejs:547]: In GatewaySwaggerResourcesProvider.java.ejs autowires SwaggerResources which is InMemorySwaggerResourcesProvider
Which Autoconfigure the Swagger Configuration where it uses the Pageable from 'org.springframework.data'

https://github.com/jhipster/jhipster/blob/main/jhipster-framework/src/main/java/io/github/jhipster/config/apidoc/SpringfoxAutoConfiguration.java


return docket
            .apiInfo(apiInfo)
            .useDefaultResponseMessages(properties.isUseDefaultResponseMessages())
            .groupName(MANAGEMENT_GROUP_NAME)
            .host(properties.getHost())
            .protocols(new HashSet<>(Arrays.asList(properties.getProtocols())))
            .forCodeGeneration(true)
            .directModelSubstitute(ByteBuffer.class, String.class)
            .genericModelSubstitutes(ResponseEntity.class)
            **.ignoredParameterTypes(Pageable.class)**
            .select()
            .paths(regex(managementContextPath + ".*"))
            .build();

Comment (mshima) [reply generators/server/templates/build.gradle.ejs:547]: You are right.
https://github.com/jhipster/jhipster-bom/blob/3489f74b0e2d25b9e1e61a9352f798cfc42776b9/jhipster-framework/src/main/java/tech/jhipster/config/apidoc/SpringfoxAutoConfiguration.java#L172

**Comment by yuvarajsanjeevi** [comment]
@mshima Please let me know what needs to be done next

**Comment by mshima** [comment]
Let’s ask @DanielFran about this hard requirement jhipster-bom has on spring-data-common.

**Comment by Tcharl** [comment]
Also I guess for pagination: `Page` is included in spring-data

File: generators/server/templates/build.gradle.ejs (hunk 1 above)
Comment (mshima) [reply generators/server/templates/build.gradle.ejs:547]: @yuvarajsanjeevi can you try to make Page conditional at jhipster-bom?
Something like:
```
        builder = docket
            .apiInfo(apiInfo)
            .useDefaultResponseMessages(properties.isUseDefaultResponseMessages())
            .groupName(MANAGEMENT_GROUP_NAME)
            .host(properties.getHost())
            .protocols(new HashSet<>(Arrays.asList(properties.getProtocols())))
            .forCodeGeneration(true)
            .directModelSubstitute(ByteBuffer.class, String.class)
            .genericModelSubstitutes(ResponseEntity.class)
        try {
          builder.ignoredParameterTypes(Class.forName("org.springframework.data.domain.Pageable"))
        } catch(error) {
        }
        return builder
            .select()
            .paths(regex(properties.getManagementIncludePattern()))
            .build();
```

Comment (yuvarajsanjeevi) [reply generators/server/templates/build.gradle.ejs:547]: @mshima 
Created Pull request in jhipster bom
https://github.com/jhipster/jhipster-bom/pull/430

Comment (mshima) [reply generators/server/templates/build.gradle.ejs:547]: Is possible to drop the dependency with this change?

Comment (mshima) [reply generators/server/templates/build.gradle.ejs]: @yuvarajsanjeevi can you add the dependency again, please?
I’ve triggered the same build and didn’t work.
We probably should make syncLdp logic optional to drop spring-data dependency.

Comment (yuvarajsanjeevi) [reply generators/server/templates/build.gradle.ejs]: @mshima Can you please share with me the build number or error details

Comment (mshima) [reply generators/server/templates/build.gradle.ejs]: Same from https://github.com/jhipster/generator-jhipster/pull/15978#issuecomment-903045764
https://github.com/hipster-labs/jhipster-daily-builds/actions/runs/1152758979

Comment (yuvarajsanjeevi) [reply generators/server/templates/build.gradle.ejs]: IMO we do not required ReactivePageableHandlerMethodArgumentResolver, ReactiveSortHandlerMethodArgumentResolver if databaseType is 'no'

Comment (yuvarajsanjeevi) [reply generators/server/templates/build.gradle.ejs]: Can you please trigger the same build again

Comment (mshima) [reply generators/server/templates/build.gradle.ejs]: done

File: generators/server/templates/src/main/java/package/config/WebConfigurer.java.ejs [hunk 2]
```diff
@@ -49,7 +49,7 @@ import org.springframework.context.annotation.Profile;
     <%_ } _%>
 import org.springframework.core.annotation.Order;
 <%_ } _%>
-<%_ if (reactive) { _%>
+<%_ if (reactive && databaseType != 'no') { _%>
```
Comment (mshima) [inline generators/server/templates/src/main/java/package/config/WebConfigurer.java.ejs:52]:
```suggestion
<%_ if (reactive && !databaseTypeNo) { _%>
```

File: generators/server/templates/src/main/java/package/config/WebConfigurer.java.ejs [hunk 3]
```diff
@@ -205,6 +205,7 @@ public class WebConfigurer implements <% if (!reactive) { %>ServletContextInitia
     }
 <%_ if (reactive) { _%>
 
+    <%_ if (databaseType != 'no') { _%>
```
Comment (mshima) [inline generators/server/templates/src/main/java/package/config/WebConfigurer.java.ejs:208]:
```suggestion
  <%_ if (!databaseTypeNo) { _%>
```

File: generators/server/templates/src/main/java/package/config/WebConfigurer.java.ejs [hunk 4]
```diff
@@ -216,6 +217,7 @@ public class WebConfigurer implements <% if (!reactive) { %>ServletContextInitia
     HandlerMethodArgumentResolver reactiveSortHandlerMethodArgumentResolver() {
         return new ReactiveSortHandlerMethodArgumentResolver();
     }
+    <%_ } _%>
```
Comment (mshima) [inline generators/server/templates/src/main/java/package/config/WebConfigurer.java.ejs:220]:
```suggestion
  <%_ } _%>
```

File: generators/server/templates/build.gradle.ejs (hunk 1 above)
Comment (mshima) [reply generators/server/templates/build.gradle.ejs]: It worked.

**Review by mshima — APPROVED** [review]
LGTM

**Comment by mshima** [comment]
@yuvarajsanjeevi congrats on the first contribution.

**Comment by yuvarajsanjeevi** [comment]
@mshima  Thank you for your great support.

**Comment by jdubois** [comment]
@yuvarajsanjeevi just to be sure, did you send this OpenCollective invoice https://opencollective.com/generator-jhipster/expenses/48874 ? (I want to be sure it's you, before paying)

**Comment by yuvarajsanjeevi** [comment]
@jdubois Yes I'm the one who raised this invoice

**Pull request merged**